claim/claim.go

```diff
@@ -151,3 +145,18 @@ func (c Claim) Validate() error {
 	// validate the Result
 	return errors.Wrap(c.Result.Validate(), "claim validation failed")
 }
+
+// ulidMutex guards the generation of ULIDs, because the use of rand
+// is not thread-safe.
+var ulidMutex sync.Mutex
+
+// ulidEntropy must be set once and reused when generating ULIDs, to guarantee
+// that each ULID is monotonically increasing.
+var ulidEntropy = ulid.Monotonic(rand.New(rand.NewSource(time.Now().UnixNano())), 0)
+
+// ULID generates a string representation of a ULID.
+func ULID() string {
+	ulidMutex.Lock()
+	defer ulidMutex.Unlock()
+	return ulid.MustNew(ulid.Timestamp(time.Now()), ulidEntropy).String()
```

Comment:
I didn't realize `ULID` could panic before now; obviously this is a separate discussion but it might be worth a breaking change to this package's API to follow the "Must_" naming convention. Perhaps `NewULID() (string, error)` and `MustNewULID() string` would be most idiomatic?

Reply:
Thanks for calling out the panic. I liked your suggestion, and made those two functions and switched any code that could use NewULID over to it. 

I'm redoing our claims code right now and once I'm done it should all use NewULID. Changing that here would have been too much though since claim.Update doesn't return an error.